ha/core/system_health/system_health.py

```diff
@@ -337,60 +337,66 @@ def process_event(self, healthevent: HealthEvent):
         """
         Process Event method. This method could be called for updating the health status.
         """
+        for retry in range(HA_MAX_RETRY):
```

Comment:
Blanket retry without knowing the cause of the exception can cause issues.
ex
If we have updated health of some components already and then we retry the whole sequence, extra updates might happens. This can cause unwanted /extra events to be sent to Hare.
If  health of any module gets updated multiple times un-necessarily, the timestamps will get modified for the same and that can lead to issues.
More such problems might also come. 

Please refer to the code from main branch in file ./ha/core/event_analyzer/watcher/watcher.py
Here the exceptions are getting handled individually. We need to do something similar ether in fault-tolerance_driver.py or in EventAnalyzer() [where we are getting those same exceptions.]

Reply:
changed as discussed and tested for every exception and found expected behaviour